src/main/java/ee/tuleva/onboarding/mandate/MandateController.java

```diff
@@ -49,27 +50,27 @@
     @ApiOperation(value = "Create a mandate")
     @RequestMapping(method = POST, value = "/mandates")
     @JsonView(MandateView.Default.class)
-    public Mandate create(@ApiIgnore @AuthenticationPrincipal User user,
+    public Mandate create(@ApiIgnore @AuthenticationPrincipal AuthenticatedPerson authenticatedPerson,
                                  @Valid @RequestBody CreateMandateCommand createMandateCommand,
                                  @ApiIgnore @Valid Errors errors) throws ValidationErrorsException {
         if (errors.hasErrors()) {
             log.info("Create mandate command is not valid: {}", errors);
             throw new ValidationErrorsException(errors);
         }
 
-        return mandateService.save(user, createMandateCommand);
+        return mandateService.save(authenticatedPerson.getUser().orElseThrow(RuntimeException::new), createMandateCommand);
```

Comment:
I think it makes sense to encapsulate the exception throwing inside the `getUser()`, so you can avoid repeating yourself with stuff like `authenticatedPerson.getUser().orElseThrow(UserNotAuthenticatedException::new)`.